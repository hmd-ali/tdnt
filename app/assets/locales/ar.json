{
  "test": "test",
  "form": {
    "errors": {
      "email": {
        "required": "البريد الإلكتروني مطلوب"
      }
    }
  }
}
