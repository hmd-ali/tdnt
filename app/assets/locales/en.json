{
  "test": "test",
  "form": {
    "errors": {
      "email": {
        "required": "Email is required"
      }
    }
  }
}
