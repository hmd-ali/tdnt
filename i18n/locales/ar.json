{
  "form": {
    "email": "بريد إلكتروني",
    "field_is_required": "{0} مطلوب.",
    "field_must_be_a_string": "يجب أن يكون {0} سلسلة.",
    "password": "كلمة المرور",
    "email_must_be_valid": "يجب أن يكون بريدًا إلكترونيًا صالحًا",
    "password_min_length": "يجب أن تتكون كلمة المرور من 8 أحرف على الأقل"
  },
  "login": "تسجيل الدخول",
  "missing_token": "الرمز مفقود أو غير صالح.",
  "logout": "تسجيل الخروج",
  "welcome_name": "مرحبًا {name}!",
  "enter_username_and_pass": "",
  "login_success": "نجاح تسجيل الدخول!",
  "invalid_username_or_password": "اسم المستخدم أو كلمة المرور غير صالحة",
  "logged_in_route": "تم تسجيل الدخول في الطريق"
}
