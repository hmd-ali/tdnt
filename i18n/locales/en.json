{
  "form": {
    "field_must_be_a_string": "{0} must be a string.",
    "field_is_required": "{0} is required.",
    "email": "Email",
    "password": "Password",
    "email_must_be_valid": "Must be a valid email",
    "password_min_length": "Password must be at least 8 characters long"
  },
  "login": "Login",
  "missing_token": "Token is missing or invalid.",
  "logout": "Logout",
  "enter_username_and_pass": "Enter your username and password",
  "welcome_name": "Welcome, {name}!",
  "invalid_username_or_password": "Invalid username or password",
  "login_success": "Login success!",
  "logged_in_route": "Logged in route",
  "back_home": "Go back home"
}
