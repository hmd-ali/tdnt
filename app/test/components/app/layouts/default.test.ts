import { mountSuspended } from "@nuxt/test-utils/runtime"
import { describe, expect, it, vi } from "vitest"
import { createTestingPinia } from "@pinia/testing"

import Default from "~/layouts/default.vue"
import { createI18n } from "vue-i18n"
import en from "~~/i18n/locales/en.json"
import ar from "~~/i18n/locales/ar.json"

const i18n = createI18n({
  locale: "en",
  messages: {
    en,
    ar,
  },
})

const stubs = ["AppLocaleSwitch"]

const mount = () =>
  mountSuspended(Default, {
    global: {
      stubs,
      plugins: [
        createTestingPinia({
          createSpy: vi.fn,
        }),
        i18n,
      ],
    },
  })

describe("layouts/default.vue", async () => {
  const locales = useNuxtApp().$i18n.locales

  it("renders the component", async () => {
    const wrapper = await mount()
    expect(wrapper.exists()).toBe(true)
  })

  describe.each(locales.value.map((l) => [l.code]))(
    "user is logged in the %s locale",
    (code) => {
      it("it renders a logout", async () => {
        const wrapper = await mount()
        const store = useUserStore()
        store.user = {
          name: "John Doe",
          email: "",
          id: "1",
          token: "",
        }
        wrapper.vm.$i18n.locale = code
        await wrapper.vm.$nextTick()
        const logoutString = i18n.global.t("logout")
        expect(wrapper.html()).toContain(logoutString)
        const welcomeString = i18n.global.t("welcome_name", {
          name: store.user.name,
        })
        expect(wrapper.text()).toContain(welcomeString)
        expect(wrapper.html()).toMatchSnapshot()
      })
    },
  )
})
