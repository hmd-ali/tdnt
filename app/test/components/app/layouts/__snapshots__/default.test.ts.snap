// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`layouts/default.vue > user is logged in the ar locale > it renders a logout 1`] = `
"<div class="container mx-auto w-full pt-20">
  <header class="fixed left-0 top-0 w-full">
    <div class="container mx-auto flex h-20 w-full items-center justify-between border-b">
      <div class="flex items-center gap-x-4">
        <p class="text-muted-foreground">مرحبًا <span class="font-medium text-primary">John Doe</span>!</p><button class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&amp;_svg]:pointer-events-none [&amp;_svg]:size-4 [&amp;_svg]:shrink-0 bg-destructive text-destructive-foreground shadow-sm hover:bg-destructive/90 h-9 px-4 py-2" type="button">تسجيل الخروج</button>
      </div>
      <app-locale-switch-stub></app-locale-switch-stub>
    </div>
  </header>
  <div></div>
</div>"
`;

exports[`layouts/default.vue > user is logged in the en locale > it renders a logout 1`] = `
"<div class="container mx-auto w-full pt-20">
  <header class="fixed left-0 top-0 w-full">
    <div class="container mx-auto flex h-20 w-full items-center justify-between border-b">
      <div class="flex items-center gap-x-4">
        <p class="text-muted-foreground">Welcome, <span class="font-medium text-primary">John Doe</span>!</p><button class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&amp;_svg]:pointer-events-none [&amp;_svg]:size-4 [&amp;_svg]:shrink-0 bg-destructive text-destructive-foreground shadow-sm hover:bg-destructive/90 h-9 px-4 py-2" type="button">Logout</button>
      </div>
      <app-locale-switch-stub></app-locale-switch-stub>
    </div>
  </header>
  <div></div>
</div>"
`;
